### PR TITLE
DEV-2209 Push PipelineValidated message when done

### PR DESCRIPTION
**Pull request by nedleitch**
This will allow us to eliminate a race condition that occurs when the lab
doesn't immediately process the snpcheck for a run, and the run completes
pipeline processing before snpcheck results are available. This is not an
exceptional case, it happens relatively frequently.

### CODE REVIEW

File: src/test/java/com/hartwig/snpcheck/SnpCheckTest.java [hunk 1]
```diff
@@ -52,6 +56,14 @@
     private static final String BARCODE = "barcode";
     private static final Sample REF_SAMPLE = new Sample().barcode(BARCODE).name("sampler");
     private static final Sample TUMOR_SAMPLE = new Sample().barcode(BARCODE).name("samplet");
+    private static final ObjectMapper OBJECT_MAPPER = new ObjectMapper();
```
Comment (pauldwolfe) [inline src/test/java/com/hartwig/snpcheck/SnpCheckTest.java:59]: bit of duplication here, maybe a shared util to create the OM?

File: src/test/java/com/hartwig/snpcheck/SnpCheckTest.java [hunk 2]
```diff
@@ -216,15 +230,50 @@ public void publishesTurquoiseEventOnCompletion() throws Exception {
...
-                new ObjectMapper().readValue(new String(pubsubMessageArgumentCaptor.getValue().getData().toByteArray()),
-                        new TypeReference<>() {
-                        });
-        assertThat(message.get("type")).isEqualTo("snpcheck.completed");
-        List<Map<String, String>> subjects = (List<Map<String, String>>) message.get("subjects");
-        assertThat(subjects.get(0).get("name")).isEqualTo("samplet");
-        assertThat(subjects.get(0).get("type")).isEqualTo("sample");
+        verify(publisher, times(2)).publish(pubsubMessageArgumentCaptor.capture());
+        boolean foundTurquoiseEvent = false;
+        for (PubsubMessage rawMessage: pubsubMessageArgumentCaptor.getAllValues()) {
```
Comment (pauldwolfe) [inline src/test/java/com/hartwig/snpcheck/SnpCheckTest.java:235]: this is a bit awkward. lets try to refactor out the loop and the if(keySet). Maybe using contains or something?

**Review by pauldwolfe — CHANGES_REQUESTED** [review]

File: src/test/java/com/hartwig/snpcheck/SnpCheckTest.java (hunk 2 above)
Comment (nedleitch) [reply src/test/java/com/hartwig/snpcheck/SnpCheckTest.java]: See what you think of what I've just pushed, the fundamental problem is that the events are essentially maps and both a bit different in structure.

File: src/test/java/com/hartwig/snpcheck/SnpCheckTest.java (hunk 1 above)
Comment (nedleitch) [reply src/test/java/com/hartwig/snpcheck/SnpCheckTest.java]: Fair enough, cleaned up.

**Pull request merged**